instchown.c

```diff
@@ -56,22 +89,30 @@ void c(char *home, char *subdir, char *file, uid_t uid, gid_t gid, int mode)
   }
   if (chmod(file,mode) == -1)
     strerr_die6sys(111,FATAL,"unable to chmod .../",subdir,"/",file,": ");
+  free(dh.s);
 }
 
 void z(char *home, char *file, int len, uid_t uid, gid_t gid, int mode)
 {
+  stralloc dh = { 0 };
+  ddhome(&dh, home);
+  home=dh.s;
   if (chdir(home) == -1)
     strerr_die4sys(111,FATAL,"unable to switch to ",home,": ");
   if (chown(file,uid,gid) == -1)
     strerr_die6sys(111,FATAL,"unable to chown ",home,"/",file,": ");
   if (chmod(file,mode) == -1)
     strerr_die6sys(111,FATAL,"unable to chmod ",home,"/",file,": ");
+  free(dh.s);
 }
 
-int main(void)
+int main(int argc, char **argv)
 {
   umask(077);
   init_uidgid();
-  hier();
+  if (argc == 2 && strcmp(argv[1],"queue") == 0)
+    hier_queue();
+  else
+    hier();
```

Comment:
Yes, because `instqueue` creates the queue files and directories, but with default permissions. `instchown queue` instead will just fix up the permissions on them.

Reply:
Feels inconsistent to say `instqueue` (no args) to install only a queue, followed by `instchown queue` to chown only a queue. Should we change `instqueue` (which has not yet been in a release) to `instpackage queue`, so the two steps work more similarly?